src/validateAtoms.ts

```diff
@@ -0,0 +1,70 @@
+import { atom, SetStateAction, WritableAtom } from 'jotai';
+import { loadable } from 'jotai/utils';
+
+import type { CommonState } from './atomWithValidate';
+
+export type Validator = <Values extends Record<string, unknown>>(
+  values: Values,
+) => Promise<Values>;
```

Comment:
```suggestion
) => Promise<unknown>;
```


and,

```tsx
type State<Values extends Record<string, unknown>> = {
  values: Values;
} & ValidatorState
```
(In the end, I'd combine it into one type)